crates/searchers/openapi.yaml

```diff
@@ -0,0 +1,584 @@
+openapi: 3.1.0
+info:
+  title: Searcher API
+  description: |
+    The API implemented by searchers interacting with the reference driver
+    implementation.
+  version: 0.0.1
+
+paths:
+  /:
+    post:
+      description: |
+        Solve the passed in auction instance.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/AuctionInstance"
+      responses:
+        200:
+          description: Auction successfully solved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AuctionResult"
+        400:
+          description: There is something wrong with the requests.
+        500:
+          description: Something went wrong when handling the request.
+
+components:
+  schemas:
+
+    Address:
+      description: |
+        20 byte Ethereum address encoded as a hex with `0x` prefix.
+      type: string
+      example: "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+
+    TokenAmount:
+      description: |
+        Amount of an ERC20 token. 256 bit unsigned integer in decimal notation.
+      type: string
+      example: "1234567890"
+
+    BigInt:
+      description: |
+        An arbitrary-precision integer value.
+      type: string
+      example: "1234567890"
+
+    Decimal:
+      description: |
+        An arbitrary-precision decimal value.
+      type: string
+      example: "13.37"
+
+    NativePrice:
+      description: |
+        The price in wei of the native token (Ether for example) to buy 10**18
+        of a token.
+      type: string
+      example: "1234567890"
+
+    Digest:
+      description: |
+        A 32-byte hex-encoded digest.
+      type: string
+      example: "0x1e66721bb1bd77d2641c77ea1d61e8abb92bf69c64fcc90c2c6ad518d1b50db1"
+
+    Token:
+      description: Information about a token relavant to the auction.
+      type: object
+      properties:
+        decimals:
+          type: integer
+          description: |
+            The ERC20.decimals value for this token.
+        symbol:
+          type: string
+          description: |
+            The ERC20.symbol value for this token.
+        externalPrice:
+          $ref: "#/components/schemas/NativePrice"
+        availableBalance:
```

Comment:
My issue with `contractBalance` is "which contract"? The token contract, settlement contract, etc.

I think the reason I like `avaibleBalance` is that this is the balance that is available for searchers to use for optimizations.

Reply:
> I think we have a dedicated term for this -> `buffer`.

Personally, I'm not a huge fan of the term. I know we've established that internally, but if you were an onboarding solver, would this make sense to you?

Reply:
Agree, `buffer` doesn't reveal much. Also the concept of internalization is not easy to explain with one word, so maybe some combination of `internal` and `protocol`... But current naming is also fine since description is there.

Reply:
Leaving it as `availableBalance` and `trusted` for now. We can come back and change this later.